Refresh repository to be used in 2022 (#5)
jenkins-infra/docker-repo-proxy: README.adoc

= docker-packaging

image:https://img.shields.io/docker/pulls/jenkinsciinfra/repo-proxy?label=jenkinsciinfra%repo-proxy&logo=docker&logoColor=white[link="https://hub.docker.com/r/jenkinsciinfra/repo-proxy"]

This repository contains everything needed to build a docker image for caching artefacts from repo.jenkins-ci.org repository to accelerate builds.

== Running the service

> Start a local Docker container (in background) based on this image with:

Reviewer: ```suggestion
Start a local Docker container (running in background) based on this image with:
```

[source,bash]
----
docker run --rm --name local-artifact-cacher -p 80:80 --detach jenkinssciinfra/repo-proxy # should add a tag in production
----

> Cached object should return 200 with HEADER 'X-Cache-Status' set to 'HIT' or 'MISS'

Reviewer: ```suggestion
Cached object should return a status code `200` with HEADER `X-Cache-Status` set to `HIT` or `MISS`
```

[source,bash]
----
curl -I http://localhost/releases/cf/pgmann/plugins/url-auth-sso/1.0/url-auth-sso-1.0-javadoc.jar
----

> Snapshot artifacts shouldn't be cached and so return 200 without HEADER 'X-Cache-Status'

Reviewer: ```suggestion
Snapshot artifacts shouldn't be cached and thus return a status code `200` without HEADER `X-Cache-Status`
```

> Snapshot artifacts shouldn't be cached and thus return a status code `200` without HEADER `X-Cache-Status`

Reviewer 2: Probably _could_ be cached, at least timestamped versions (since we are not caching `maven-metadata.xml`), but fine—rarely used. Artifacts from the `incrementals` repository definitely _can_ be cached, and it is important to verify that: while they are used similarly to snapshots, from a technical standpoint they are releases (for our purposes: once uploaded to a particular URL, will never be overwritten with different content).

Author: Thanks Jessie. i've opened https://github.com/jenkins-infra/docker-repo-proxy/issues/7 to track this proposal.

[source,bash]
----
curl -I http://localhost/snapshots/jenkins/repository/1.1-SNAPSHOT/repository-1.1-20140112.102452-1-javadoc.jar
----

> maven-metadata.xml object should return 200 without HEADER 'X-Cache-Status'

Reviewer: ```suggestion
maven-metadata.xml object should return a status code `200` without HEADER `X-Cache-Status`
```

[source,bash]
--
curl -I http://localhost/snapshots/com/deveo/plugins/deveo/maven-metadata.xml
--

[IMPORTANT]

> All requests should return 200 OK

Reviewer: ```suggestion
All requests should return a status code `200` "OK"
```

[IMPORTANT]
--

> Cached requests contain HEADER 'X-Cache-Status':
> 
> * Set to HIT if response get from cache
> * Set to MISS if response didn't get from cache

Reviewer: ```suggestion
Cached requests contain HEADER `X-Cache-Status` set to:

* `HIT` if it got its response from cache
* `MISS` if it didn't got its response from cache
```
